source/dub/dub.d

```diff
@@ -285,11 +311,48 @@ class Dub {
 			}
 		}
 
+		// Override user + local package path from /etc/dub/settings.json
+		// Then continues loading local settings from these folders. (keeping
+		// global /etc/dub/settings.json settings intact)
+		//
+		// Don't use it if either $DPATH or $DUB_HOME are set.
+		if (!overrideUserSettings) {
+			if (m_config.userSettings.length) {
+				m_dirs.userSettings = NativePath(m_config.userSettings.expandEnvironmentVariables);
+
+				overrideUserSettings = true;
+			} else if (m_config.dubHome.length) {
+				m_dirs.userSettings = NativePath(m_config.dubHome.expandEnvironmentVariables);
+
+				overrideUserSettings = true;
+			}
+		}
```

Comment:
The circular dependency is IMO weird. I need a bit of time to think about it.

Reply:
it's a little weird in the code, but I think it's great for user flexibility.

A use case here: package maintainer creates global settings.json in `/var/lib/dub` which then specifies where the user-overridable config is located. On linux distros that have read-only filesystems with some special paths for writable things that might for example be useful. Or for users using non-standard paths like storing all dub things on a separate flash drive or something that just isn't mounted in /etc.

Additionally it's possible to make the config only per-user and use an environment variable like `$HOME` inside the settings file so each user can have different dub settings.